### DISCUSSION TITLE
How does the provisioning profile creation work in regards to entitlements?

### DISCUSSION THREAD

**Discussion by tealshift** [fastlane/fastlane#17277] [Q&A] [answered]
I use a script I run to configure my Xcode project to one of a set of configurations. It changes the bundle ID and everything it needs to produce a unique app before fastlane does its magic. 

One of the things that changes between configuration is the entitlements file's App Group ID.
When I run Match to generate provisioning profiles, I expect the entitlements associated with those profiles to agree with the entitlements files in my project. However to my surprise, it's generating profiles with the App Group ID of my _base configuration_. I have inspected the profiles in the `~/Library/MobileDevice/Provisioning Profiles` folder and confirmed this is the case. 

The entitlements file is configured like:
```
<key>com.apple.security.application-groups</key>
<array>
	<string>group.com.company.myapp</string> // <--- This is what I am configuring
</array>
```

Is there some place that this App Group ID is being cached and reused for new profiles instead of reading the ID from my entitlements files? I have had some success with Match generating with the correct app group ID, but I think it's because I had Xcode running, perhaps updating some cached value for me after I ran my app configuration.

**Comment by tealshift** [comment]
Also, this might be related: I've been updating my entitlements files manually with PlistBuddy, but Fastlane offers this action `update_app_group_identifiers`. However I can't get it to work.
It just shouts: 
> Error setting value '/path/to/entitlements_file.entitlements' for option 'entitlements_file'
You passed invalid parameters to 'update_app_group_identifiers'

I thought maybe this action could somehow fix this problem I have, but I can't get it to accept my path argument, no matter what it is. Anyone have luck with this action?

**Comment by tealshift** [comment, marked as answer]
I figured it out! The App ID registered with Apple was pointing to the wrong App Group ID! It doesn't matter what your entitlement file is set to when it generates the profiles.

**Comment by arehart31** [comment]
Amazing post, I was searching for the same topic [UPSers](https://www.upsers.run/).
